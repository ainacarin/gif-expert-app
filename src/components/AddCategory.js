import React, { useState } from "react";
import PropTypes from 'prop-types';
import "./AddCategory.css";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');
  const [error, setError] = useState(null);


  const resetError = () => setError('');

  const handleChangeInput = (event) => {
    resetError();
    setInputValue(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    // validation no empty input
    if (inputValue.trim().length > 1) {
      setCategories((oldCategories) => [...oldCategories, inputValue]);
      // clean input
      setInputValue("");
    } else{
      setError('Error value');
    }
  };

  return (
    <>
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          value={inputValue}
          onChange={handleChangeInput}
          placeholder="Write new character"
        ></input>
        {error && <p>{error}</p>}
      </form>
    </>
  );
};

// Proptypes definition
AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}