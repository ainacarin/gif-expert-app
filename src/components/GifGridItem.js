import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ( { id, title, url } ) => {

    return (
        <div id={id} className='card animate__animated animate__fadeIn'>
            <p>{title}</p>
            <img src={url} alt={title} />
        </div>
    );

};

GifGridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}