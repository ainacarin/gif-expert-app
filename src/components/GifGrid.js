import React, { useState, useEffect } from "react";
import { getGifs } from "../api/giphyApi";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

    // exect only when dependencies are modified
    useEffect(() => {
      getGifs( category ).then(setImages);
    }, [category]);
    //   // exect only once time, the first time component
    //   useEffect(() => {
    //     getGifs( category ).then( setImages );
    //   }, []);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};
