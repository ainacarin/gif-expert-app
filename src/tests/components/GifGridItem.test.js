import React from 'react';
import { shallow } from 'enzyme';

import { GifGridItem } from '../../components/GifGridItem';

describe('Group Tests GifGridItem', () => {

    const url = 'https://localhost/item.jpg';
    const title = 'Title item';
    const classFadeIn = 'animate__fadeIn';

    const wrapper = shallow( <GifGridItem title={title} url={url}/> );
    
    test('GifGridItem: ok display', () => {


        expect( wrapper ).toMatchSnapshot();
    });

    test('GifGridItem: title', () => {
        const pElem = wrapper.find('p');
        expect( pElem.text() ).toBe( title )
    })

    test('GifGridItem: img url & alt', () => {
        const img = wrapper.find('img');
        const { src, alt } = img.props();
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    })

    test('GifGridItem: class animate__fadeIn', () => {
        const div = wrapper.find('div');
        const classes = div.prop('className');
        expect(div.hasClass(classFadeIn)).toBe( true );
        expect(classes.includes(classFadeIn)).toBe( true );

    })

})