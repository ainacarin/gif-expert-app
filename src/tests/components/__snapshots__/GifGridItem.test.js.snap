// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group Tests GifGridItem GifGridItem: ok display 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <p>
    Title item
  </p>
  <img
    alt="Title item"
    src="https://localhost/item.jpg"
  />
</div>
`;
